draft-ietf-rats-corim.md

```diff
@@ -1780,70 +1787,118 @@ Regardless of the specific integrity protection method used, the Evidence's inte
+Input Conceptual Messages, whether Endorsements, Reference Values, Evidence, or Policies, are transformed to an internal representation that is based on ECTs ({{sec-ir-cm}}).
 
 The following mapping conventions apply to all forms of input transformation:
-The `e` field is populated with a Target Environment identifier.
-The `c` field is populated with the measurements collected by an Attesting Environment.
-The `a` field is populated with the identity of the entity that asserted (e.g., signed) the Evidence.
-The `ns` field is populated with the namespace context if supplied. For example, the Attester's manufacturer may have a URI that identifies the manufacturing series, family or architecture.
-The `cm` field is set based on the type of Conceptual Message inputted or to be outputed.
+
+> * The `environment` field is populated with a Target Environment identifier.
+> * The `element-list` field is populated with the measurements collected by an Attesting Environment.
+> * The `authority` field is populated with the identity of the entity that asserted (e.g., signed) the Conceptual Message.
+> * The `cmtype` field is set based on the type of Conceptual Message inputted or to be outputed.
+> * The `profile` field is set based on the `corim-map` `profile` value.
 
 #### Appraisal Context Construction
 
 All of the extracted and validated tags are loaded into an *appraisal context*.
 The Appraisal Context contains an internal representation of the inputted Conceptual Messages.
 The selected tags are mapped to an internal representation, making them suitable for appraisal processing.
 
-[^issue] https://github.com/ietf-rats-wg/draft-ietf-rats-corim/issues/96
-
 #### Reference Triples Transformation {#sec-ref-trans}
 
 {:rtt-enum: counter="foo" style="format Step %d."}
 
 {: rtt-enum}
 * An available `rv` list entry ({{sec-ir-ref-val}}) is allocated.
-The Reference Values Triple ({{sec-comid-triple-refval}}) `environment-map` is copied to the `environment-map` for both the `rv` `condition` and `rv` `addition` ECTs.
 
-* For each `measurement-map` entry in the measurements list, the i<sup>th</sup> `measurement-map` is copied to the i<sup>th</sup> `element-map` in the `elements` list of the `rv` `addition` ECT.
+* The ECT `cmtype` of the `ev` entry is set to `reference-values`.
 
-* The issuer of the Endorsement conceptual message is copied to the `ev` `addition` ECT authority field.
+* The Reference Values Triple ({{sec-comid-triple-refval}}) `environment-map` is copied to the `environment-map` for both the `rv` `condition` and `rv` `addition` ECTs.
 
-* If the Endorsement conceptual message has a profile, the profile is copied to the `ev` `addition` ECT profile field.
+* For each `measurement-map` entry in the measurements list, the i<sup>th</sup> `measurement-map` is copied to the i<sup>th</sup> `element-map` in the `element-list` of the `rv` `addition` ECT.
```

Comment:
I didn't think order would matter, given that mkey is the distinguishing feature, and not the position in the element-list.

Reply:
> I didn't think order would matter, given that mkey is the distinguishing feature, and not the position in the element-list.

The processing sections pay attention to mkey semantics. This is saying that the transformation isn't expected to change position semantics. It may be harmless but it's not my intention to do that here.

There is a yet-to-be-discussed issue regarding what the processing requirements are if mkey is omitted and multiples of measurement-map exist. The CDDL permits this. I think it's reasonable for the verifier use array position to disambiguate. The RIM author decides which approach is better. Otherwise, we have to put normative next banning this condition.

Reply:
Issue #303 created and tracking annotations added.